lib/ASRouterTargeting.jsm

```diff
@@ -172,6 +172,28 @@ function sortMessagesByTargeting(messages) {
   });
 }
 
+/**
+ * Sort messages in descending order based on the value of `priority`
+ * Messages with no `priority` are ranked lowest (even after a message with
+ * priority 0).
+ */
+function sortMessagesByPriority(messages) {
+  return messages.sort((a, b) => {
+    if (isNaN(a.priority) && isNaN(b.priority)) {
+      return 0;
+    }
+    if (!isNaN(a.priority) && isNaN(b.priority)) {
+      return -1;
+    }
+    if (isNaN(a.priority) && !isNaN(b.priority)) {
+      return 1;
+    }
+
+    // Descending order; higher priority comes first
+    return a.priority > b.priority ? -1 : 1;
```

Comment:
If they are equal, this will return 1. Technically, it should return 0 but I guess the end result will be the same

Reply:
updated. Now the order of messages with same priority is left unchanged.